### PR TITLE
parquet-tools: init at 0.2.9

### PR DESCRIPTION
**Pull request by cpcloud**
###### Motivation for this change

Add `parquet-tools` to nixpkgs.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -1785,6 +1785,8 @@ with pkgs;
 
   pacparser = callPackage ../tools/networking/pacparser { };
 
+  parquet-tools = with python3Packages; toPythonApplication parquet-tools;
```
[the remark this answers is not in the document]
Comment (cpcloud) [reply pkgs/top-level/all-packages.nix:1788]: Should I call this something else?

File: pkgs/development/python-modules/parquet-tools/default.nix [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    sha256 = "0aw0x7lhagp4dwis09fsizr7zbhdpliav0ns5ll5qny7x4m6rkfy";
+  };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace 'thrift = "^0.13.0"' 'thrift = "*"' \
+      --replace 'halo = "^0.0.29"' 'halo = "*"'
+  '';
+
+  nativeBuildInputs = [ poetry ];
```
[the remark this answers is not in the document]
Comment (cpcloud) [reply pkgs/development/python-modules/parquet-tools/default.nix:38]: Ok, we can update this package once that PR lands. In the meantime, can we merge this PR?

File: pkgs/top-level/all-packages.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (veprbl) [reply pkgs/top-level/all-packages.nix:1788]: Actually, I just saw it claims to be compatible. So we probably do want to ship it instead of parquet-tools.

**Review by veprbl — APPROVED** [review]
LGTM

File: pkgs/development/python-modules/parquet-tools/default.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/development/python-modules/parquet-tools/default.nix:38]: Can we fetchpatch the patch? This shouldn't be forgotten with the next update.

[the remark this answers is not in the document]
Comment (cpcloud) [reply pkgs/development/python-modules/parquet-tools/default.nix]: done

**Comment by veprbl** [comment]
Could you please squash all changes?

**Comment by cpcloud** [comment]
@veprbl Yup, squashing now.

**Comment by veprbl** [comment]
I believe test_excute_simple fails to accept 6.0.0 instead of 6.0.0 for arrow-cpp version.

@GrahamcOfBorg build parquet-tools

**Comment by cpcloud** [comment]
@SuperSandro2000 @veprbl I've fixed the tests.

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 142562` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>parquet-tools (python39Packages.parquet-tools)</li>
    <li>python38Packages.parquet-tools</li>
  </ul>
</details>

**Comment by veprbl** [comment]
I get a following runtime error on darwin:
```
 parquet-tools 
Traceback (most recent call last):
  File "/nix/store/a2vf0h1sfzsfbm42bi524qzqyxykdy62-python3.9-parquet-tools-0.2.9/bin/.parquet-tools-wrapped", line 6, in <module>
    from parquet_tools.cli import main
  File "/nix/store/5k8ry64svlp23hpypbzpp8ic5i1q2abi-python3.8-parquet-tools-0.2.9/lib/python3.8/site-packages/parquet_tools/cli.py", line 2, in <module>
    from parquet_tools.commands import show, csv, inspect
  File "/nix/store/5k8ry64svlp23hpypbzpp8ic5i1q2abi-python3.8-parquet-tools-0.2.9/lib/python3.8/site-packages/parquet_tools/commands/show.py", line 6, in <module>
    import pandas as pd
  File "/nix/store/mprs63r6bq9a65wa2r3a47lla98xlvg3-python3.8-pandas-1.3.3/lib/python3.8/site-packages/pandas/__init__.py", line 16, in <module>
    raise ImportError(
ImportError: Unable to import required dependencies:
numpy: 

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy C-extensions failed. This error can happen for
many reasons, often due to issues with your setup or how NumPy was
installed.

We have compiled some common reasons and troubleshooting tips at:

    https://numpy.org/devdocs/user/troubleshooting-importerror.html

Please note and check the following:

  * The Python version is: Python3.9 from "/nix/store/lrw7ifwnpz7rskfkphq9mj9fwm5i9ldc-python3-3.9.6/bin/python3.9"
  * The NumPy version is: "1.21.2"

and make sure that they are the versions you expect.
Please carefully study the documentation linked above for further help.

Original error was: No module named 'numpy.core._multiarray_umath'
```

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -5465,6 +5465,8 @@ in {
 
   parquet = callPackage ../development/python-modules/parquet { };
 
+  parquet-tools = callPackage ../development/python-modules/parquet-tools { };
```
Comment (veprbl) [inline pkgs/top-level/python-packages.nix:5468]: Sorry for not realizing this earlier, but this piece of software is unlikely to be useful as a library. So it should be using `buildPythonApplication`, the `pythonPackages.parquet-tools` attribute should be removed and the nix file moved out of pkgs/development/python-modules.

Comment (cpcloud) [reply pkgs/top-level/python-packages.nix:5468]: Thanks! Fixing this now.

**Review by veprbl — APPROVED** [review]

**Comment by cpcloud** [comment]
Hm, it looks like for some reason gRPC fails to build on `aarch64-linux` and jemalloc (a dependency of `arrow-cpp`) fails to build on `aarch64-darwin`. @veprbl any idea why that might be?

**Comment by veprbl** [comment]
Let's discuss aarch64-darwin failures in #146823 or file separate issues

**Pull request merged**